comid/integregisters.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package comid
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/veraison/swid"
+)
+
+// IRegisterIndex is the interface to hold register index
+type IRegisterIndex interface{}
+
+// IntegReg holds the Integrity Registers
+type IntegReg struct {
+	Register map[IRegisterIndex]Digests
+}
+
+func NewIntegRegister() (*IntegReg, error) {
+	reg := make(map[IRegisterIndex]Digests)
+	return &IntegReg{Register: reg}, nil
+}
+
+// AddDigests allows inserting a series of Digests at a specific RegisterIndex
+func (i *IntegReg) AddDigests(index IRegisterIndex, digests Digests) error {
+	if len(digests) == 0 {
+		return fmt.Errorf("no digests to add")
+	}
+	for _, digest := range digests {
+		if err := i.AddDigest(index, digest); err != nil {
+			return fmt.Errorf("unable to add Digest: %w", err)
+		}
+	}
+	return nil
+}
+
+// AddDigest allows inserting a Digest at a specific RegisterIndex
+func (i *IntegReg) AddDigest(index IRegisterIndex, digest swid.HashEntry) error {
+
+	if i.Register == nil {
+		return fmt.Errorf("no register to add digest")
+	}
+	switch t := index.(type) {
+	case string, uint, uint64:
+		i.Register[t] = append(i.Register[t], digest)
+	default:
+		return fmt.Errorf("unexpected type for index: %T", t)
+	}
+	return nil
+}
+
+// RemoveIndex removes an Index from the Register
+func (i *IntegReg) RemoveIndex(index IRegisterIndex) error {
```

Comment:
This doesn't seem to provide any useful functionality.

Reply:
Removed it for now!